src/DIRAC/DataManagementSystem/Agent/FTS3Agent.py

```diff
@@ -638,14 +631,10 @@ def execute(self):
 
     @staticmethod
     def __sendAccounting(ftsJob):
-        """prepare and send DataOperation to AccountingDB
-
-        :param ftsJob: the FTS3Job from which we send the accounting info
-        """
-
-        dataOp = DataOperation()
-        dataOp.setStartTime(fromString(ftsJob.submitTime))
-        dataOp.setEndTime(fromString(ftsJob.lastUpdate))
-
-        dataOp.setValuesFromDict(ftsJob.accountingDict)
-        dataOp.delayedCommit()
+        dataOpSender = DataOperationSender()
+        dataOpSender.sendData(
+            ftsJob.accountingDict,
+            delayedCommit=True,
```

Comment:
Considering that there's a delayed commit, I think it would make sense to:

1) create the `dataOpSender` object (that would be a `self.dataOpSender`) inside the method `beginExecution`. This would ensure that a new `self.dataOpSender` object is created every time there's a new cycle starting. 

and

2) add this method to the agent (that would be executed every time the agent is finishing the cycle):

```python
    def endExecution(self):
        self.dataOpSender.concludeSending()
```

Reply:
Sounds great! Even though I believe we don't need to have a endExecution method since, having `delayedCommit=True` means that we already commit all the additions done in the script